### PR TITLE
Update transformer modules

### PR DESCRIPTION
**Pull request by KPatr1ck**

### CODE REVIEW

File: modules/text/language_model/bert-base-cased/module.py [hunk 1]
```diff
@@ -20,20 +19,22 @@
 import paddle.nn.functional as F
 
 from paddlehub import BertTokenizer
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/module.py]: 可以尝试从paddlenlp调用相应的tokenizer

File: modules/text/language_model/bert-base-cased/module.py [hunk 2]
```diff
@@ -75,6 +93,16 @@ def forward(self, input_ids, token_type_ids=None, position_ids=None, attention_m
                 acc = self.metric.update(correct)
                 return probs, loss, acc
             return probs
+        elif self.task == 'token-cls':
+            logits = result
+            token_level_probs = F.softmax(logits, axis=2)
+            if labels is not None:
+                labels = paddle.to_tensor(labels).unsqueeze(-1)
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/module.py]: 为什么labels 还需要再转换成tensor，labels应该是从dataloader中返回的，已经是tensor了。

File: modules/text/language_model/bert-base-chinese/README.md [hunk 3]
```diff
@@ -13,29 +14,35 @@ $ hub install bert-base-chinese==2.0.0
...
-* `task`： 任务名称，可为`sequence_classification`。
+* `task`： 任务名称，可为`seq-cls`(文本分类任务，原来的`sequence_classification`在未来会被弃用)或`token-cls`(序列标注任务)。
 * `load_checkpoint`：使用PaddleHub Fine-tune api训练保存的模型参数文件路径。
 * `label_map`：预测时的类别映射表。
+* `num_classes`：分类任务的类别数，如果指定了`label_map`，此参数可不传，默认2分类。
+* `**kwargs`：用户额外指定的关键字字典类型的参数。
 
 ```python
 def predict(
     data,
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-chinese/README.md:35]: texts，同理其他文档注意修改

File: modules/text/language_model/bert-base-chinese/README.md [hunk 4]
```diff
@@ -13,29 +14,35 @@ $ hub install bert-base-chinese==2.0.0
...
     max_seq_len=128,
     batch_size=1,
-    use_gpu=False)
+    use_gpu=False
+)
 ```
 
 **参数**
 
 * `data`： 待预测数据，格式为\[\[sample\_a\_text\_a, sample\_a\_text\_b\], \[sample\_b\_text\_a, sample\_b\_text\_b\],…,\]，其中每个元素都是一个样例，
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-chinese/README.md:44]: texts，同理其他文档注意修改

File: modules/text/language_model/bert-base-cased/module.py [hunk 5]
```diff
@@ -75,6 +93,16 @@ def forward(self, input_ids, token_type_ids=None, position_ids=None, attention_m
...
+                loss = self.criterion(logits, labels)
+                correct = self.metric.compute(token_level_probs, labels)
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/module.py]: metric的处理 需要注意 label padding的情况。参考 这里的处理方式：https://github.com/PaddlePaddle/models/blob/release/2.0-beta/PaddleNLP/examples/named_entity_recognition/msra_ner/run_msra_ner.py#L316

File: modules/text/language_model/bert-base-cased/README.md [hunk 6]
```diff
@@ -14,23 +14,29 @@ $ hub install bert-base-cased==2.0.0
...
-* `task`： 任务名称，可为`sequence_classification`。
+* `task`： 任务名称，可为`seq-cls`(文本分类任务，原来的`sequence_classification`在未来会被弃用)或`token-cls`(序列标注任务)。
 * `load_checkpoint`：使用PaddleHub Fine-tune api训练保存的模型参数文件路径。
 * `label_map`：预测时的类别映射表。
+* `num_classes`：分类任务的类别数，如果指定了`label_map`，此参数可不传，默认2分类。
+* `**kwargs`：用户额外指定的关键字字典类型的参数。
 
 ```python
 def predict(
     data,
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/README.md:35]: texts，同理注意其他文档的修改

File: modules/text/language_model/bert-base-cased/README.md [hunk 7]
```diff
@@ -46,6 +52,8 @@ def predict(
 ```python
 def get_embedding(
     texts,
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/README.md]: texts，同理注意其他文档的修改

File: modules/text/language_model/bert-base-cased/module.py [hunk 8]
```diff
@@ -44,17 +45,34 @@ def __init__(
...
+            )
+        if task == 'seq-cls':
+            self.model = BertForSequenceClassification.from_pretrained(pretrained_model_name_or_path='bert-base-cased', num_classes=self.num_classes, **kwargs)
+            self.criterion = paddle.nn.loss.CrossEntropyLoss()
+            self.metric = paddle.metric.Accuracy()
+        elif task == 'token-cls':
+            self.model = BertForTokenClassification.from_pretrained(pretrained_model_name_or_path='bert-base-cased', num_classes=self.num_classes, **kwargs)
             self.criterion = paddle.nn.loss.CrossEntropyLoss()
-            self.metric = paddle.metric.Accuracy(name='acc_accumulation')
+            self.metric = paddle.metric.Accuracy()
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/module.py]: 序列标注任务指标一般使用f1 score

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
以上意见，其他类似地方注意修改

File: modules/text/language_model/bert-base-cased/README.md [hunk 9]
```diff
@@ -68,24 +78,26 @@ def get_embedding(
 import paddlehub as hub
 
 data = [
-    '这个宾馆比较陈旧了，特价的房间也很一般。总体来说一般',
-    '怀着十分激动的心情放映，可是看着看着发现，在放映完毕后，出现一集米老鼠的动画片',
-    '作为老的四星酒店，房间依然很整洁，相当不错。机场接机服务很好，可以在车上办理入住手续，节省时间。',
+    ['这个宾馆比较陈旧了，特价的房间也很一般。总体来说一般'],
```
Comment (Steffy-zxf) [inline modules/text/language_model/bert-base-cased/README.md:81]: 这是适用于 序列标注任务的 预测数据？为什么 要改变成 预测数据为 list的list ，建议和之前版本参数数据格式保持统一。

File: modules/text/language_model/bert-base-cased/module.py (hunk 1 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/module.py]: 目前已在ernie_tiny修改，reivew通过后其他模块同步修改

File: modules/text/language_model/bert-base-cased/module.py (hunk 2 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/module.py]: 已改

File: modules/text/language_model/bert-base-cased/module.py (hunk 5 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/module.py]: 数据集中针对padding的label特殊处理，并在cross entropy中加入对应的ignore_label

File: modules/text/language_model/bert-base-cased/module.py (hunk 8 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/module.py]: 与paddlenlp同步，使用paddlenlp.metrics.ChunkEvaluator计算f1 score

File: modules/text/language_model/bert-base-cased/README.md (hunk 9 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/README.md:81]: 这个README.md里的代码片无法运行，查看了demo/text_classification里的示例代码，用的是List[List[str]]，所以才把这里的改了

File: paddlehub/datasets/base_nlp_dataset.py [hunk 10]
```diff
@@ -276,6 +289,7 @@ def __init__(self,
             label_file=label_file,
             label_list=label_list)
 
+        self.chunk_scheme = chunk_scheme.value
```
Comment (Steffy-zxf) [inline paddlehub/datasets/base_nlp_dataset.py]: 需要检查下 用户传入的 chunk_scheme 是否符合预期。

Comment (KPatr1ck) [reply paddlehub/datasets/base_nlp_dataset.py]: 已加强制类型校验

File: modules/text/language_model/bert-base-chinese/README.md (hunk 3 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-chinese/README.md:35]: https://github.com/PaddlePaddle/PaddleHub/pull/1137
统一改成data

File: modules/text/language_model/bert-base-chinese/README.md (hunk 4 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-chinese/README.md:44]: https://github.com/PaddlePaddle/PaddleHub/pull/1137
统一改成data

File: modules/text/language_model/bert-base-cased/README.md (hunk 6 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/README.md:35]: https://github.com/PaddlePaddle/PaddleHub/pull/1137
统一改成data

File: modules/text/language_model/bert-base-cased/README.md (hunk 7 above)
Comment (KPatr1ck) [reply modules/text/language_model/bert-base-cased/README.md]: https://github.com/PaddlePaddle/PaddleHub/pull/1137
统一改成data

**Review by Steffy-zxf — APPROVED** [review]
LGTM
